Userland/Libraries/LibWeb/HTML/HTMLInputElement.cpp

```diff
@@ -2368,4 +2368,9 @@ HTMLInputElement::ValueAttributeMode HTMLInputElement::value_attribute_mode() co
     VERIFY_NOT_REACHED();
 }
 
+bool HTMLInputElement::is_child_node_selectable(DOM::Node* node) const
+{
+    return !is_button() && (!node || node->parent_element() != m_placeholder_element);
```

Comment:
This doesn't handle the case where `node == m_placeholder_element`. Maybe check instead for `m_placeholder_element->is_inclusive_ancestor_of(node)`, which will catch that, and any future situation where the placeholder has a more complicated tree.